### PR TITLE
Fix Firebase distribution workflow

### PR DESCRIPTION
**Pull request by b-onc**
...and switch to 1st party "cancel concurrent workflows" solution

#skip_changelog

### CODE REVIEW

File: .github/workflows/distribute-test-build.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ on:
       - main
   workflow_dispatch:
 
+concurrency: 
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (nuno-vieira) [inline .github/workflows/distribute-test-build.yml:11]: Finally 🎉

**Review by adamrushy — APPROVED** [review]
Great 👍

File: .github/workflows/smoke-checks.yml [hunk 2]
```diff
@@ -10,6 +10,10 @@ on:
   release:
     types:
       - created
+
+concurrency: 
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (nuno-vieira) [inline .github/workflows/smoke-checks.yml:16]: I know we are not using them right now, but maybe let's add this to the full checks as well? 🤔

Comment (b-onc) [reply .github/workflows/smoke-checks.yml:16]: Full checks are not run on pushes to main, this is not required for it. same applies to testflight

**Pull request merged**